ophyd/v2/core.py

```diff
@@ -607,3 +616,21 @@ async def describe_configuration(self) -> Dict[str, Descriptor]:
 
     async def read_configuration(self) -> Dict[str, Reading]:
         return await merge_gathered_dicts(sig.read() for sig in self._conf_signals)
+
+
+KT = TypeVar("KT")
+
+
+class DeviceDict(Dict[KT, Device], Device):
```

Comment:
@tacaswell this started as a `DeviceVector(d1, d2, d3)`, and became `DeviceDict({1: d1, 2: d2, 3: d3})` so we could be 1-indexed and gappy. I think using a dict is correct, but should we allow any typed keys, or should we only support int keys and call it a `DeviceVector`? If so, should we support both `DeviceVector(d1, d2, d3)` and `DeviceVector({1: d1, 2: d2, 3: d3})` to construct the same thing?